src/api/PDFDocument.ts

```diff
@@ -1264,6 +1265,20 @@ export default class PDFDocument {
     return dataUri ? `data:application/pdf;base64,${base64}` : base64;
   }
 
+  findPageForRef(ref: PDFRef): PDFPage {
+    const pages = this.getPages();
+    for (let idx = 0, len = pages.length; idx < len; idx++) {
+      const page = pages[idx];
+      const annotations = page.node.Annots();
+
+      if (annotations?.indexOf(ref) !== undefined) {
+        return page;
+      }
+    }
+
+    throw new Error(`Could not find page for PDFRef ${ref}`);
```

Comment:
I'd like to avoid throwing an error here in order to keep this method as flexible as possible. It may be used in scenarios where the caller isn't sure whether a ref lookup will succeed or not.
```suggestion
    return undefined;
```